service/rtc/sfu.go

```diff
@@ -550,9 +567,15 @@ func (s *Server) InitSession(cfg SessionConfig, closeCb func() error) error {
 			call.mut.Lock()
 
 			us.mut.Lock()
-			us.outScreenTracks[trackIdx] = outScreenTracks
-			us.remoteScreenTracks[trackIdx] = remoteTrack
-			us.screenRateMonitors[trackIdx] = rm
+			if isScreen {
+				us.outScreenTracks[trackIdx] = outTracks
+				us.remoteScreenTracks[trackIdx] = remoteTrack
+				us.screenRateMonitors[trackIdx] = rm
+			} else {
+				us.outVideoTracks[trackIdx] = outTracks
+				us.remoteVideoTracks[trackIdx] = remoteTrack
+				us.videoRateMonitors[trackIdx] = rm
+			}
```

Comment:
Is this duplication that could be fixed?

Reply:
Cool, I just want to make sure we're not missing something actually beneficial. Definitely don't needlessly generalize.